app/models/katello/host_collection.rb

```diff
@@ -30,8 +30,13 @@ class HostCollection < Katello::Model
     scoped_search :on => :organization_id, :complete_value => true
 
     def validate_max_content_hosts
+      if new_record?
+        if (!unlimited_content_hosts) && (systems.size.to_i > max_content_hosts.to_i)
+          errors.add :base,  _("You cannot have more than #{max_content_hosts} content host(s) associated with host collection #{name}.")
+        end
+      end
```

Comment:
Wondering if this would be better in a separate method to help break up the code here.